main.go

```diff
@@ -1,15 +1,30 @@
 package main
 
 import (
-	"github.com/hashicorp/terraform-plugin-sdk/plugin"
-	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"context"
+	"flag"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
 	"github.com/mercari/terraform-provider-spinnaker/spinnaker"
 )
 
 func main() {
-	plugin.Serve(&plugin.ServeOpts{
-		ProviderFunc: func() terraform.ResourceProvider {
-			return spinnaker.Provider()
-		},
-	})
+	var debugMode bool
+
+	flag.BoolVar(&debugMode, "debuggable", false, "set to true to run the provider with support for debuggers like delve")
+	flag.Parse()
+
+	if debugMode {
+		err := plugin.Debug(context.Background(), "registry.terraform.io/mercari/spinnaker",
+			&plugin.ServeOpts{
+				ProviderFunc: spinnaker.Provider,
+			})
+		if err != nil {
+			log.Println(err.Error())
+		}
+	} else {
+		plugin.Serve(&plugin.ServeOpts{
+			ProviderFunc: spinnaker.Provider})
+	}
```

Comment:
[note]
This is new implementation for Debuggable Provider Binary:
https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html#support-for-debuggable-provider-binaries